### ISSUE TITLE
Memory exhaustion on large queries

### ISSUE THREAD

**Issue by BrunoWinck** [WP-API/WP-API#1646]
I manage large sites. Some have been build over years by bloggers and have over 4000 posts, some are filled automatically but provide a good use case for robustness. Recaps is one of those: http://recaps.kneaver.com/pkmchat-wol-for-organizations-w-buffer/

Using the new API I was looking forward better performance and scalability. 

filter[nopaging] cause server side memory exhaustion above 1500
I used paging as follow in a while loop.

filter[offset] = From;
filter[posts_per_page] = Count;
filter[orderby] = post_modified post_date";
filter[order] = "DESC" 

From += Count;

With Count = 200 I fail at 1500 as well
With 50 I can go to 2000 (strange impact)
So the further I want to go, the smallest slices. Not very productive.
Eventually I can't go behind 2000.

Of course i could increase the memory on server side but it would just increase the limit.
IMHO the query should be very minimal for every records before offset and complete afterward.

Also I'm querying with context=edit but still have the rendered content which can be considerably larger. Waste of memory and bandwith. I would suggest context being an array and allowing edit and display or edit only.

I keep a cache on client side but for initial connection I have no other choice that fetching them all.

**Comment by danielbachhuber** [comment]
> filter[nopaging] cause server side memory exhaustion above 1500

This is because your server isn't configured to prepare 1500 posts in one request. The point of pagination is to be able to break your request into smaller pieces.

**Comment by BrunoWinck** [comment]
Daniel you didn't read my post :( I explained that I tried pagination down to 50)
I'm a bit suprised that each and every of my issues are downplayed to remove the essence of what I wrote and then closed without consideration.
The same code worked with xmlrpc for years.
I'm not a newbie in IT. I know how to confgure a server, I know PHP and I know WordPress. i've been working on APIs for years.
Now I've seen this tactic many times and silencing issues don't grow quality as much as addressing them.
If anything it just means you don't want any feedback.
I'm not going to insist but you made my opinion on the quality to expect of this API.
